packages/cloud-functions/src/cico/rampWebhook.ts

```diff
@@ -0,0 +1,50 @@
+import crypto from 'crypto'
+import stableStringify from 'fast-json-stable-stringify'
+import * as functions from 'firebase-functions'
+import { readFileSync } from 'fs'
+import { RAMP_KEY } from '../config'
+import { saveTxHashProvider } from '../firebase'
+import { Provider } from './Provider'
+
+const rampKey = readFileSync(`./config/${RAMP_KEY}`).toString()
+
+function verifyRampSignature(signature: string | undefined, body: any) {
+  if (!signature || !body) {
+    return false
+  }
+
+  const verifier = crypto.createVerify('sha256')
+  verifier.update(Buffer.from(stableStringify(body)))
+  return verifier.verify(rampKey, signature, 'base64')
+}
+
+const RAMP_SIGNATURE_HEADER = 'X-Body-Signature'
+const RAMP_RELEASED = 'RELEASED'
+const RAMP_RELEASING = 'RELEASING'
+
+export const rampWebhook = functions.https.onRequest((request, response) => {
+  if (verifyRampSignature(request.header(RAMP_SIGNATURE_HEADER), request.body)) {
+    const { type, purchase } = request.body
+
+    console.info('Received Ramp webhook', type, purchase)
+    if (type === RAMP_RELEASED || type === RAMP_RELEASING) {
```

Comment:
Are there events we are purposefully ignoring?

Reply:
There are many intermediate states: 

<img width="575" alt="Captura de Pantalla 2021-03-26 a la(s) 09 39 37" src="https://user-images.githubusercontent.com/6062888/112632684-3173d980-8e17-11eb-9e5d-75501f984206.png">

In fact, we probably don't need the `RELEASING`, `RELEASED` should be enough. We're listening for expired and cancelled on the analytics PR

Reply:
I agree.

For the analytics PR, I'd air on storing more data rather than less. It might be helpful to understand at what step in the process are taking the longest for different providers. We probably need some amount of standardization across providers so we can make apples-for-apples comparisons